testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -10417,3 +10422,151 @@ fn microblock_miner_multiple_attempts() {
+    let receiver_contract_principal: PrincipalData =
+        QualifiedContractIdentifier::new(receiver_stx_addr.into(), receiver_contract_name).into();
+    let receiver_standard_principal: PrincipalData =
+        StandardPrincipalData::from(receiver_stx_addr).into();
+
+    let peg_wallet_address =
+        address::PoxAddress::Addr32(false, address::PoxAddressType32::P2TR, [0; 32]);
+
+    let (mut conf, _) = neon_integration_test_conf();
+
+    let epoch_2_05 = 210;
+    let epoch_2_1 = 215;
+
+    let mut epochs = core::STACKS_EPOCHS_REGTEST.to_vec();
+    epochs[1].end_height = epoch_2_05;
+    epochs[2].start_height = epoch_2_05;
+    epochs[2].end_height = epoch_2_1;
+    epochs[3].start_height = epoch_2_1;
+
+    conf.node.mine_microblocks = false;
+    conf.burnchain.max_rbf = 1000000;
+    conf.miner.first_attempt_time_ms = 5_000;
+    conf.miner.subsequent_attempt_time_ms = 10_000;
+    conf.node.wait_time_for_blocks = 0;
+
+    conf.burnchain.epochs = Some(epochs);
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    let mut run_loop = neon::RunLoop::new(conf.clone());
+
+    btcd_controller
+        .start_bitcoind()
+        .ok()
+        .expect("Failed starting bitcoind");
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    btc_regtest_controller.bootstrap_chain(216);
+
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let run_loop_coordinator_channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // Let's send a Peg-in op.
+    let peg_in_op_standard = PegInOp {
+        recipient: receiver_standard_principal,
+        peg_wallet_address: peg_wallet_address.clone(),
+        amount: 1337,
+        memo: Vec::new(),
+        txid: Txid([0u8; 32]),
+        vtxindex: 0,
+        block_height: 0,
+        burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+    };
+
+    // Let's send a Peg-in op with a contract principal.
+    let peg_in_op_contract = PegInOp {
+        recipient: receiver_contract_principal,
+        peg_wallet_address,
+        amount: 1337,
+        memo: Vec::new(),
+        txid: Txid([1u8; 32]),
+        vtxindex: 0,
+        block_height: 0,
+        burn_header_hash: BurnchainHeaderHash([0u8; 32]),
+    };
+
+    let mut miner_signer = Keychain::default(conf.node.seed.clone()).generate_op_signer();
+    assert!(
+        btc_regtest_controller
+            .submit_operation(
+                StacksEpochId::Epoch21,
+                BlockstackOperationType::PegIn(peg_in_op_standard.clone()),
+                &mut miner_signer,
+                1
+            )
+            .is_some(),
+        "Peg-in operation should submit successfully"
+    );
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let parsed_peg_in_op_standard = {
```

Comment:
This is fine for now, but this information should also be obtainable from the event observer.  Please open an issue so we can remember to go back and add it.  Thanks!

Reply:
Sure, I created #3554